core/services/vrf/delegate.go

```diff
@@ -389,17 +389,26 @@ func (lsn *listener) runLogListener(unsubscribes []func(), minConfs uint32) {
 
 func (lsn *listener) handleLog(lb log.Broadcast, minConfs uint32) {
 	if v, ok := lb.DecodedLog().(*solidity_vrf_coordinator_interface.VRFCoordinatorRandomnessRequestFulfilled); ok {
+		if should := lsn.shouldProcessLog(lb); !should {
```

Comment:
if we get a block after a log has been sent but before its been marked consumed, we'll try to mark it consumed twice and the second one will fail (observed that in prod)